kbase/narrative
# DATAUP-641 parameter errors 8 - checkboxes

# Description of PR purpose/changes

Updates checkboxes to better respond to errors with spreadsheet values. Now, they:
* show an error with the wrong value, and set to a default value
* have a dismiss button for the error
* reset the error when the checkbox is clicked

This one's a bit of an outlier - as of now, it's not really possible for a checkbox to start with an error value. It will automatically replace the wrong value with the default from the parameter spec. These specs are also set (when imported by `common/sdk`) to only have 0 / 1 as values. So in this case, the "error" doesn't mean that the app won't run, it automatically gets fixed. This also means that the error display code is part of the checkbox itself, not of the parent widget. So, a little different from the other input widgets.


# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-641

# Testing Instructions
* Details for how to test the PR:
* Make a BI cell with an xsv file, and set what should be a checkbox value to anything but 0 or 1

# Dev Checklist:


# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
